Initial NoShip drop of Mp Metrics 5.0

Channyboy opened the pull request:
New Features:
- io.openliberty.mpMetrics-5.0.feature new mpMetrics-5.0 feature
- io.openliberty.org.eclipse.microprofile.metrics-5.0.feature for the api
- io.openliberty.mpMetrics-5.0-monitor-1.0.feature for vendor metrics
- io.openliberty.restfulWS3.1-mpMetrics-monitor1.0.feature for REST metrics
-

Updated Features:
- io.openliberty.restfulWSDefaultExceptionMapper-1.0.feature support mpMetrics 5 and restfulWS-3.1
-

New:
- io.openliberty.smallrye.metrics project contains the smallrye metrics code
- io.openliberty.micrometer project contains micrometer dependencies (core + target prom registry + its deps)
- io.openliberty.restfulWS.mpMetrics.filter project contains code to handle mpmetrics' REST metrics
- io.openliberty.microprofile.metrics.internal.5.0 bulk of code proxying and interfacing with SmallRye Metrics code and providing access to other bundles that may need MetricRegistry + REST handler
- io.openliberty.microprofile.metrics.internal.private.5.0 contains bundle for secured /metric connection
- io.openliberty.microprofile.metrics.internal.public.5.0 contains bundle for unsecured /metric connection
- io.openliberty.microprofile.metrics.internal.5.0.monitor contains logic interfacing with monitor-1.0 to create vendor metrics, due to changes in mpmetrics the behaviour is changed

Other :
- Updated the mpMetrics metatype.xml to accomdate LibraryRef. This used for mpMetrics-5.0 where uesr can provide their own Micrometer libraries

#build

Following were uploaded to artifactory:
- microprofile-metrics-api-tck-5.0.0-RC2  
- io.micrometer:micrometer-core:1.9.3     
- io.micrometer:micrometer-registry-prometheus:1.9.3  
- org.hdrhistogram:HdrHistogram:2.1.12    
- io.prometheus:simpleclient:0.15.0
- io.prometheus:simpleclient_common:0.15.0
- io.prometheus:simpleclient_tracer_common:0.15.0
- io.prometheus:simpleclient_tracer_otel:0.15.0
- io.prometheus:simpleclient_tracer_otel_agent:0.15.0
- io.smallrye:smallrye-metrics:5.0.0-M2

- ***org.latencyutils:LatencyUtils:2.0.3   is needed but was apparently already uploaded in 2018.

Related to https://github.com/OpenLiberty/open-liberty/issues/20018

Channyboy committed:
Initial NoShip drop of Mp Metrics 5.0

New Features:
- io.openliberty.mpMetrics-5.0.feature new mpMetrics-5.0 feature
- io.openliberty.org.eclipse.microprofile.metrics-5.0.feature for the
api
- io.openliberty.mpMetrics-5.0-monitor-1.0.feature for vendor metrics
- io.openliberty.restfulWS3.1-mpMetrics-monitor1.0.feature for REST
metrics
-

Updated Features:
- io.openliberty.restfulWSDefaultExceptionMapper-1.0.feature support
mpMetrics 5 and restfulWS-3.1
-

New:
- io.openliberty.smallrye.metrics project contains the smallrye metrics
code
- io.openliberty.micrometer project contains micrometer
dependencies (core + target prom registry + its deps)
- io.openliberty.restfulWS.mpMetrics.filter project contains code to handle mpmetrics' REST metrics
- io.openliberty.microprofile.metrics.internal.5.0 bulk of code proxying
and interfacing with SmallRye Metrics code and providing access to other
bundles that may need MetricRegistry + REST handler
- io.openliberty.microprofile.metrics.internal.private.5.0 contains
bundle for secured /metric connection
- io.openliberty.microprofile.metrics.internal.public.5.0 contains
bundle for unsecured /metric connection
- io.openliberty.microprofile.metrics.internal.5.0.monitor contains
logic interfacing with monitor-1.0 to create vendor metrics, due to
changes in mpmetrics the behaviour is changed

Other :
- Updated the mpMetrics metatype.xml to accomdate LibraryRef. This used
for mpMetrics-5.0 where uesr can provide their own Micrometer libraries

tevans78 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/bnd.bnd:
```diff
@@ -0,0 +1,76 @@
+	io.openliberty.org.eclipse.microprofile.metrics.5.0;version=latest,\
+	com.ibm.websphere.appserver.spi.kernel.service;version=latest,\
+	com.ibm.websphere.appserver.spi.httptransport;version=latest,\
+	com.ibm.websphere.javaee.servlet.3.1;version=latest,\
+	com.ibm.websphere.appserver.spi.ssl;version=latest,\
+	com.ibm.websphere.javaee.cdi.2.0;version=latest, \
+	com.ibm.ws.cdi.interfaces;version=latest,\
```
These need to be the Jakarta EE 10 versions ... but if the PB passes like this then get it merged in and fix it in a follow up.

jhanders34 replied:
```suggestion
	io.openliberty.jakarta.cdi.4.0;version=latest, \
	com.ibm.ws.cdi.interfaces.jakarta;version=latest,\
```

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/public/mpMetrics-5.0/resources/l10n/com.ibm.websphere.appserver.mpMetrics-5.0.properties:
```diff
@@ -0,0 +1,17 @@
+#NLS_MESSAGEFORMAT_NONE
+#
+
+description=This feature provides support for the Eclipse MicroProfile Metrics 5.0 specification for enterprise Java.
```
This file is not needed.  io.openliberty.mpMetrics-5.0.properties already exists and is the right name.

jhanders34 commented on dev/com.ibm.ws.microprofile.metrics.common/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -18,4 +18,8 @@ mpMetrics=MicroProfile Metrics
+authentication.desc=Indicates whether or not users should provide user name and password when accessing metrics endpoint.
+
+libraryRef=Shared Library
+libraryRef$Ref=Shared library reference
```
What is libraryRef$Ref?

Channyboy replied:
Copied this from other features who also use LibrayRef.
A lot of the other metatype.properties  that use the LibraryRef seem to have it.
e.g.
`com.ibm.ws.cloudant`

jhanders34 replied:
Something we can talk about at the externals review next week if this gets in before that.

jhanders34 commented on dev/io.openliberty.micrometer/.classpath:
```diff
@@ -0,0 +1,8 @@
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

jhanders34 commented on dev/io.openliberty.micrometer/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,290 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=11
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=11
```

tevans78 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/original.bnd:
```diff
@@ -0,0 +1,14 @@
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+
+Bundle-Name: MicroProfile Metrics 5.0
```
I think you need a `transformed.bnd` as well?

jhanders34 commented on dev/io.openliberty.micrometer/bnd.bnd:
```diff
@@ -0,0 +1,53 @@
+Bundle-Name: io.openliberty.micrometer
+Bundle-SymbolicName: io.openliberty.micrometer
+Bundle-Description: Micrometer libraries and necessary dependencies, version ${bVersion}
+
```
```suggestion

javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/.classpath:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

jhanders34 commented on dev/io.openliberty.micrometer/.classpath:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,11 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=11
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=11
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/bnd.bnd:
```diff
@@ -0,0 +1,51 @@
+Bundle-Description: MicroProfile Metrics Monitor Bridge Auto Feature, version ${bVersion}
+
+bVersion=1.0
+
```
```suggestion

javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/.classpath:
```diff
@@ -0,0 +1,8 @@
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,378 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=11
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=11
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/bnd.bnd:
```diff
@@ -0,0 +1,76 @@
+-buildpath: \
+    org.eclipse.osgi;version=latest,\
+	com.ibm.websphere.org.osgi.core;version=latest,\
+	com.ibm.ws.app.manager.wab;version=latest,\
```
```suggestion
	com.ibm.ws.app.manager.wab.jakarta;version=latest,\
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/bnd.bnd:
```diff
@@ -0,0 +1,76 @@
+bVersion=1.0.${libertyBundleMicroVersion}
+
+-sub: *.bnd
+
```
```suggestion

javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/public/mpMetrics-5.0/io.openliberty.mpMetrics-5.0.feature:
```diff
@@ -16,9 +16,10 @@ Subsystem-Name: MicroProfile Metrics 5.0
- io.openliberty.microprofile.metrics.internal.public
+ io.openliberty.io.smallrye.common.jakarta, \
+ com.ibm.ws.cdi.interfaces.jakarta, \
+ io.openliberty.microprofile.metrics.internal.5.0.jakarta, \
```
```suggestion
 io.openliberty.microprofile.metrics.internal.5.0, \
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/bnd.bnd:
```diff
@@ -0,0 +1,76 @@
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0.${libertyBundleMicroVersion}
+
+-sub: *.bnd
```
```suggestion
Bundle-Name: MicroProfile Metrics 5.0
Bundle-SymbolicName: io.openliberty.microprofile.metrics.internal.5.0
Bundle-Description: MicroProfile Metrics Common, version ${bVersion}
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/bnd.bnd:
```diff
@@ -0,0 +1,76 @@
+	javax.interceptor; version=!,\
+	javax.enterprise.context.spi; version=!,\
+	org.eclipse.microprofile.config; version="[3.0,4)",\
+	*
```
```suggestion
Import-Package: \
	jakarta.enterprise.inject; version="[4.0,5)",\
	jakarta.enterprise.util; version="[4.0,5)",\
	jakarta.annotation; version="[2.1,3)",\
	jakarta.interceptor; version="[2.1,3)",\
	jakarta.enterprise.context.spi; version="[4.0,5)",\
	org.eclipse.microprofile.config; version="[3.0,4)",\
	*
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/bnd.bnd:
```diff
@@ -0,0 +1,76 @@
+	com.ibm.ws.logging;version=latest,\
+	com.ibm.ws.rest.handler;version=latest,\
+	com.ibm.ws.classloading;version=latest,\
+	com.ibm.ws.webcontainer;version=latest,\
```
```suggestion
	com.ibm.ws.webcontainer.jakarta;version=latest,\
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/bnd.bnd:
```diff
@@ -0,0 +1,76 @@
+	io.openliberty.org.eclipse.microprofile.metrics.5.0;version=latest,\
+	com.ibm.websphere.appserver.spi.kernel.service;version=latest,\
+	com.ibm.websphere.appserver.spi.httptransport;version=latest,\
+	com.ibm.websphere.javaee.servlet.3.1;version=latest,\
```
```suggestion
	io.openliberty.jakarta.servlet.6.0;version=latest,\
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/original.bnd:
```diff
@@ -0,0 +1,14 @@
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+
```
delete this file

tevans78 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/cdi/adapters/LegacyMetricsExtensionAdapter.java:
```diff
@@ -0,0 +1,169 @@
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+import javax.enterprise.event.Observes;
+import javax.enterprise.inject.spi.AfterDeploymentValidation;
+import javax.enterprise.inject.spi.BeanManager;
+import javax.enterprise.inject.spi.BeforeBeanDiscovery;
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+import javax.enterprise.inject.spi.ProcessManagedBean;
+import javax.enterprise.inject.spi.WithAnnotations;
+
+import org.eclipse.microprofile.metrics.annotation.Counted;
+import org.eclipse.microprofile.metrics.annotation.Gauge;
+import org.eclipse.microprofile.metrics.annotation.Timed;
```
Can't mix'n'match javax and jakarta (used by the annotations)

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/metrics/SmallryeMetricsCDIMetadata.java:
```diff
@@ -0,0 +1,275 @@
+import java.util.Map;
+import java.util.Set;
+
+import javax.enterprise.inject.spi.Extension;
```
```suggestion
import jakarta.enterprise.inject.spi.Extension;
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/metrics/SmallryeMetricsCDIMetadata.java:
```diff
@@ -0,0 +1,275 @@
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.metrics;
```
Not sure why did this package with just one class.  Seems can do it in one of the internal packages also in this bundle.

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/BaseMetricsRESTProxyServlet.java:
```diff
@@ -0,0 +1,92 @@
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
```
```suggestion
import jakarta.servlet.ServletContext;
import jakarta.servlet.ServletException;
import jakarta.servlet.http.HttpServlet;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import jakarta.servlet.http.HttpSession;
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/MetricRESTHandler.java:
```diff
@@ -0,0 +1,157 @@
+import java.util.Map;
+import java.util.stream.Stream;
+
+import javax.servlet.http.HttpServletResponse;
```
```suggestion
import jakarta.servlet.http.HttpServletResponse;
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/impl/SharedMetricRegistries.java:
```diff
@@ -0,0 +1,107 @@
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.microprofile.metrics50.internal.impl;
```
Again weird to have a package for just 1 class.  Why not in the main internal package?

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/cdi/adapters/LegacyMetricsExtensionAdapter.java:
```diff
@@ -0,0 +1,169 @@
+import javax.enterprise.inject.spi.Extension;
+import javax.enterprise.inject.spi.ProcessAnnotatedType;
+import javax.enterprise.inject.spi.ProcessManagedBean;
+import javax.enterprise.inject.spi.WithAnnotations;
```
```suggestion
import jakarta.enterprise.event.Observes;
import jakarta.enterprise.inject.spi.AfterDeploymentValidation;
import jakarta.enterprise.inject.spi.BeanManager;
import jakarta.enterprise.inject.spi.BeforeBeanDiscovery;
import jakarta.enterprise.inject.spi.Extension;
import jakarta.enterprise.inject.spi.ProcessAnnotatedType;
import jakarta.enterprise.inject.spi.ProcessManagedBean;
import jakarta.enterprise.inject.spi.WithAnnotations;
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricProducerAdapter.java:
```diff
@@ -0,0 +1,146 @@
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
+import javax.inject.Inject;
```
```suggestion
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Produces;
import jakarta.enterprise.inject.spi.InjectionPoint;
import jakarta.inject.Inject;
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricRegistryProducerAdapter.java:
```diff
@@ -0,0 +1,85 @@
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Default;
+import javax.enterprise.inject.Produces;
+import javax.enterprise.inject.spi.InjectionPoint;
```
```suggestion
import jakarta.enterprise.context.ApplicationScoped;
import jakarta.enterprise.inject.Default;
import jakarta.enterprise.inject.Produces;
import jakarta.enterprise.inject.spi.InjectionPoint;
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0/transformed.bnd:
```diff
@@ -0,0 +1,15 @@
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/transform.props
```
delete this file.

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=11
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=11
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/bnd.bnd:
```diff
@@ -0,0 +1,53 @@
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
```
```suggestion

javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/bnd.bnd:
```diff
@@ -0,0 +1,53 @@
+WS-TraceGroup: METRICS
+
+-buildpath: \
+	com.ibm.websphere.javaee.servlet.3.1;version=latest,\
```
```suggestion
	io.openliberty.jakarta.servlet.6.0;version=latest,\
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/resources/WEB-INF/web.xml:
```diff
@@ -0,0 +1,74 @@
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<!DOCTYPE web-app PUBLIC "-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN" "http://java.sun.com/dtd/web-app_2_3.dtd">
```
Since this is only for EE10 / MP 6, might as well make this the servlet 6 web.xml instead of servlet 3

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.public.5.0/.classpath:
```diff
@@ -0,0 +1,8 @@
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.public.5.0/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,12 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=11
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=11
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.public.5.0/bnd.bnd:
```diff
@@ -0,0 +1,53 @@
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
+
```
```suggestion
javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"

```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.public.5.0/bnd.bnd:
```diff
@@ -0,0 +1,53 @@
+
+-buildpath: \
+	com.ibm.websphere.javaee.servlet.3.1;version=latest,\
+	io.openliberty.org.eclipse.microprofile.metrics.3.0;version=latest,\
```
```suggestion
	io.openliberty.jakarta.servlet.6.0;version=latest,\
	io.openliberty.org.eclipse.microprofile.metrics.5.0;version=latest,\
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.public.5.0/resources/WEB-INF/web.xml:
```diff
@@ -0,0 +1,27 @@
+    Contributors:
+        IBM Corporation - initial API and implementation
+ -->
+<!DOCTYPE web-app PUBLIC "-//Sun Microsystems, Inc.//DTD Web Application 2.3//EN" "http://java.sun.com/dtd/web-app_2_3.dtd">
```
Again this should be servlet 6.0 format instead of servlet 3.0

jhanders34 commented on dev/io.openliberty.org.eclipse.microprofile/metrics.5.0.bnd:
```diff
@@ -0,0 +1,26 @@
+#*******************************************************************************
+-include ~../cnf/resources/bnd/bundle.props
+bVersion=1.0
+
```
```suggestion

javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/.classpath:
```diff
@@ -0,0 +1,7 @@
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="con" path="aQute.bnd.classpath.container"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,11 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=11
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=11
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/bnd.bnd:
```diff
@@ -0,0 +1,74 @@
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props,~../cnf/resources/bnd/transform.props
+bVersion=1.0
+
```
```suggestion

javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/bnd.bnd:
```diff
@@ -0,0 +1,74 @@
+
+-buildpath: \
+	com.ibm.ws.classloading,\
+	com.ibm.websphere.javaee.annotation.1.1;version=latest,\
```
```suggestion
	io.openliberty.jakarta.annotation.2.1;version=latest,\
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/bnd.bnd:
```diff
@@ -0,0 +1,74 @@
+	com.ibm.ws.monitor,\
+	com.ibm.websphere.javaee.servlet.3.0;version=latest,\
+	com.ibm.websphere.javaee.jaxrs.2.0,\
+	io.openliberty.jakarta.restfulWS.3.0, \
```
```suggestion
	io.openliberty.jakarta.servlet.6.0;version=latest,\
	io.openliberty.jakarta.restfulWS.3.1, \
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/bnd.bnd:
```diff
@@ -0,0 +1,74 @@
+	io.openliberty.org.eclipse.microprofile.metrics.5.0;version=latest,\
+	com.ibm.ws.jaxrs.defaultexceptionmapper;version=latest, \
+	com.ibm.ws.microprofile.metrics.common;version=latest,\
+	com.ibm.ws.jaxrs.2.0.common
```
```suggestion
	com.ibm.ws.jaxrs.defaultexceptionmapper.jakarta;version=latest, \
	com.ibm.ws.microprofile.metrics.common;version=latest,\
	io.openliberty.org.jboss.resteasy.common.ee10
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/src/io/openliberty/restfulws/mpmetrics/MetricsRestfulWsEMCallbackImpl.java:
```diff
@@ -0,0 +1,148 @@
+import java.util.Collections;
+import java.util.Map;
+
+import javax.ws.rs.container.ResourceInfo;
```
```suggestion
import jakarta.ws.rs.container.ResourceInfo;
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/src/io/openliberty/restfulws/mpmetrics/RestfulWsMonitorFilter.java:
```diff
@@ -0,0 +1,298 @@
+import javax.ws.rs.container.ContainerResponseFilter;
+import javax.ws.rs.container.ResourceInfo;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.ext.Provider;
```
```suggestion
import jakarta.annotation.PostConstruct;
import jakarta.annotation.PreDestroy;
import jakarta.ws.rs.container.ContainerRequestContext;
import jakarta.ws.rs.container.ContainerRequestFilter;
import jakarta.ws.rs.container.ContainerResponseContext;
import jakarta.ws.rs.container.ContainerResponseFilter;
import jakarta.ws.rs.container.ResourceInfo;
import jakarta.ws.rs.core.Context;
import jakarta.ws.rs.ext.Provider;
```

jhanders34 commented on dev/io.openliberty.smallrye.metrics/.classpath:
```diff
@@ -0,0 +1,8 @@
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

jhanders34 commented on dev/io.openliberty.smallrye.metrics/.settings/org.eclipse.jdt.core.prefs:
```diff
@@ -0,0 +1,290 @@
+org.eclipse.jdt.core.compiler.debug.sourceFile=generate
+org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
+org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
+org.eclipse.jdt.core.compiler.source=1.8
```
```suggestion
org.eclipse.jdt.core.compiler.codegen.targetPlatform=11
org.eclipse.jdt.core.compiler.codegen.unusedLocal=preserve
org.eclipse.jdt.core.compiler.compliance=11
org.eclipse.jdt.core.compiler.debug.lineNumber=generate
org.eclipse.jdt.core.compiler.debug.localVariable=generate
org.eclipse.jdt.core.compiler.debug.sourceFile=generate
org.eclipse.jdt.core.compiler.problem.assertIdentifier=error
org.eclipse.jdt.core.compiler.problem.enumIdentifier=error
org.eclipse.jdt.core.compiler.source=11
```

jhanders34 commented on dev/io.openliberty.smallrye.metrics/bnd.bnd:
```diff
@@ -0,0 +1,34 @@
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props
+bVersion=1.0.${libertyBundleMicroVersion}
+
```
```suggestion

javac.source: 11
javac.target: 11

Require-Capability: osgi.ee; filter:="(&(osgi.ee=JavaSE)(version=11))"
```

jhanders34 commented on dev/io.openliberty.smallrye.metrics/resources/META-INF/services/javax.ws.rs.ext.Providers:
```diff
@@ -0,0 +1 @@
+io.smallrye.metrics.jaxrs.JaxRsMetricsFilter
```
This file should be called jakarta.ws.rs.ext.Providers

jhanders34 commented on dev/io.openliberty.smallrye.metrics/bnd.bnd:
```diff
@@ -0,0 +1,34 @@
+Bundle-SymbolicName: io.openliberty.smallrye.metrics
+Bundle-Description: MicroProfile Metrics SmallRye Implementation, version ${bVersion}
+
+src: src, resource
```
The resources are not added to the bundle so what is the point of the files in the resources directory?

jhanders34 replied:
My point is you are missing an -includeresources instruction to add the META-INF/services directory.

jhanders34 commented on dev/io.openliberty.smallrye.metrics/.classpath:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="src"/>
```
```suggestion
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/.classpath:
```diff
@@ -0,0 +1,8 @@
+<classpath>
+	<classpathentry kind="src" path="src"/>
+	<classpathentry kind="src" path="resources"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-1.8"/>
```
```suggestion
	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-11"/>
```

dmuelle commented on dev/com.ibm.ws.microprofile.metrics.common/resources/OSGI-INF/l10n/metatype.properties:
```diff
@@ -18,4 +18,8 @@ mpMetrics=MicroProfile Metrics
 authentication=Enable authentication 
-authentication.desc=Indicates whether or not users should provide user name and password when accessing metrics endpoint.
\ No newline at end of file
+authentication.desc=Indicates whether or not users should provide user name and password when accessing metrics endpoint.
```
```suggestion
authentication.desc=Indicates whether users must provide a username and password when they access the metrics endpoint.
```

Channyboy replied:
@dmuelle This metatype.properties definition already exists for existing mpMetrics features. I think it shows the line as "new" since new lines were added to the _old last line_. If you want to update the description for this, we can open another issue?

dmuelle replied:
no worries, I just reviewed it because it came up in our queue. If Libby tags a PR for ID review unnecessarily, you can skip it by adding the `ID review not needed` label.

dmuelle reviewed (changes requested):
ID review

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/bnd.bnd:
```diff
@@ -0,0 +1,51 @@
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props,~../cnf/resources/bnd/transform.props
```
```suggestion
-include= ~../cnf/resources/bnd/bundle.props
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/bnd.bnd:
```diff
@@ -0,0 +1,51 @@
+bVersion=1.0
+
+Import-Package: \
+	javax.enterprise.*;version="[1.1,3)",\
+	org.eclipse.microprofile.metrics; version="[4.0.0,6.0.0)",\
+	javax.ws.rs.container;resolution:=optional,\
```
```suggestion
	jakarta.enterprise.*;version="[4.0,5)",\
	org.eclipse.microprofile.metrics; version="[5.0.0,6.0.0)",\
	jakarta.ws.rs.container;resolution:=optional,\
```

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/bnd.bnd:
```diff
@@ -0,0 +1,51 @@
+	com.ibm.wsspi.org.osgi.service.component.annotations,\
+	com.ibm.ws.anno;version=latest, \
+	com.ibm.ws.org.osgi.annotation.versioning;version=latest, \
+	com.ibm.ws.jaxrs.defaultexceptionmapper;version=latest, \
+	com.ibm.websphere.javaee.jaxrs.2.0;version=latest,\
```
```suggestion
	com.ibm.ws.jaxrs.defaultexceptionmapper.jakarta;version=latest, \
	io.openliberty.jakarta.restfulWS.3.1;version=latest,\
```

jhanders34 commented on dev/io.openliberty.restfulWS.mpMetrics.filter/bnd.bnd:
```diff
@@ -0,0 +1,74 @@
+# Contributors:
+#     IBM Corporation - initial API and implementation
+#*******************************************************************************
+-include= ~../cnf/resources/bnd/bundle.props,~../cnf/resources/bnd/transform.props
```
```suggestion
-include= ~../cnf/resources/bnd/bundle.props
```

Channyboy committed:
Apply suggestions from code review

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

Channyboy committed:
Furthere Jakarta changes needed in bnd files, remove original.bnd and transform.bnd, rename to javax to jakarta.ws.rs.ext.Providers

Channyboy commented:
#build

Channyboy committed:
Adddress feature junit tests, change fatures to full for no ship and proper name

Channyboy commented:
#build

Channyboy commented:
Build will fail w.r.t fault tolerance's auto feature with metrics:

MicroProfile Metrics 5.0 removes the `register` methods in `MetricRegistry` and the `MetricType` class is removed.

Potential soln?
https://github.com/Channyboy/open-liberty/commit/746efda4121144a4df4c5f29080a827ce3fc7c53#diff-ca6f72e69245b435ef792fad16dd6952d86e1603fffd16d062582d3ac2cc4dcc

Existing `dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.mpFaultTolerance4.0-metrics.feature` 
 will remove mpMetrics-5.0

New : dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.mpFaultTolerance4.0-metrics5.0.feature
is an auto feature for mpMetrics-5.0 and mpFaultTolerance-4.0

New  `io.openliberty.microprofile.faulttolerance.4.0.internal.metrics.5.0` bundle
This gets rid of `MetricType` usage in the `MetricRecorder` class and the gauge method in `MetricRecorder`  accepts the `Supplier<Long>` as a parameter and uses the `MetricRegistry.gauge(Metadata, Supplier, Tags)` method for gauge registration.

dmuelle reviewed (approved):
ID review not needed, approving so as not to block

Channyboy committed:
remove mpMetrics-5.0 from the io.openliberty.mpFaultTolerance4.0-metrics auto-feature

Channyboy commented:
#build

Channyboy committed:
Address rest of PR review comments.

- Move SmallryeMetricsCDIMetadata and SharedMetricRegistries into the
internal package.
- Updat web.xml to us JEE 6 Servlet schema
- remove unused rsrc ref in bnd.bnd

Channyboy commented:
#build

Channyboy committed:
remove unused proprety file

Channyboy committed:
Apply suggestions from code review pt 2

One org.eclipse.jdt.core.prefs missed from first batch commit

Co-authored-by: Jared Anderson <jhanders@us.ibm.com>

Channyboy commented:
#build

Channyboy commented:
@jim-krueger 
As discussed before, updated dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restfulWSDefaultExceptionMapper-1.0.feature
To also factor in `mpMetrics-5.0` and `restfulWS-3.1`

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.mpMetrics-5.0-monitor-1.0.feature:
```diff
@@ -0,0 +1,10 @@
+-include= ~${workspace}/cnf/resources/bnd/feature.props
+symbolicName=io.openliberty.mpMetrics-5.0-monitor-1.0
+visibility=private
+IBM-Provision-Capability: \
```
mpMetrics-5.0 depends on monitor-1.0 so this auto feature is not needed and the bundle should be moved into the mpMetrics-5.0 feature file or better yet take the content of the bundle and move it to the existing mpmetrics 5.0 bundle.

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/auto/io.openliberty.restfulWS3.1-mpMetrics-monitor1.0.feature:
```diff
@@ -0,0 +1,11 @@
+symbolicName=io.openliberty.restfulWS3.1-mpMetrics-monitor1.0
+Manifest-Version: 1.0
+IBM-Provision-Capability: \
+  osgi.identity; filter:="(&(type=osgi.subsystem.feature)(osgi.identity=com.ibm.websphere.appserver.monitor-1.0))", \
```
```suggestion
```

Since monitor is already enabled by mpMetrics-5.0 no need to have it in the auto feature's list of features.

jhanders34 commented on dev/com.ibm.websphere.appserver.features/visibility/public/mpMetrics-5.0/io.openliberty.mpMetrics-5.0.feature:
```diff
@@ -16,9 +16,10 @@ Subsystem-Name: MicroProfile Metrics 5.0
- io.openliberty.microprofile.metrics.internal.private, \
- io.openliberty.microprofile.metrics.internal.public
+ io.openliberty.io.smallrye.common.jakarta, \
+ com.ibm.ws.cdi.interfaces.jakarta, \
```
```suggestion
```
Already in cdi-4.0 so not needed to be explicitly listed for this feature as well.

jhanders34 commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/bnd.bnd:
```diff
@@ -0,0 +1,55 @@
+	jakarta.ws.rs.container;resolution:=optional,\
+	*
+
+Export-Package: io.openliberty.microprofile.metrics.internal.monitor
```
Usually packages with internal in their name are not exported.  This explains why you had a separate package for those classes that were reference by other bundles.

Channyboy replied:
Created Issue https://github.com/OpenLiberty/open-liberty/issues/22819 which covers _all_ MP Metric bundles (of prior versions as well)

jhanders34 reviewed (approved):
There are some small things that still remain, but I am going to approve so you can proceed, but those things should be resolved in your next PR.

jim-krueger commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/internal/MappingTable.java:
```diff
@@ -0,0 +1,182 @@
+//			{ "base", "REST.request", "Total Requests and Response Time", "REST.request.description", SIMPLE_TIMER, MetricUnits.NANOSECONDS, "RequestCount", null, null, "ResponseTimeDetails", "total" }
+//		};
+//
+//		mappingTable.put("WebSphere:type=REST_Stats,name=*", restTable);
```
If there is a reason to keep this commented out code please add a comment.  Otherwise delete it.

Channyboy committed:
Skip running MP 6 set of features for `MetricRemovalTest` in `com.ibm.ws.microprofile.faulttolerance.cdi_fat_metrics` and skip the MP 6 set of features for io.openliberty.microprofile.faulttolerance.4.0.internal_fat_tck

Channyboy commented:
#build

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/.classpath:
```diff
@@ -0,0 +1,7 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Per https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-naming-guidelines, should the project be named as `io.openliberty.microprofile.metrics.5.0.internal.monitor`?

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/MonitorMetricsHandler.java:
```diff
@@ -0,0 +1,216 @@
+		        MBeanServerNotification mbsn = (MBeanServerNotification) notification;
+	            String objectName = mbsn.getMBeanName().toString();
+		        if(MBeanServerNotification.REGISTRATION_NOTIFICATION.equals(mbsn.getType())) {
+		            Tr.debug(tc, "MBean Registered [", objectName + "]");
```
Unconditional debug statement

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/MonitorMetricsHandler.java:
```diff
@@ -0,0 +1,216 @@
+		            	register(objectName, data);
+		            }
+		        } else if(MBeanServerNotification.UNREGISTRATION_NOTIFICATION.equals(mbsn.getType())) {
+		        	Tr.debug(tc, "MBean Unregistered [" + objectName + "]");
```
Unconditional debug statement

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/MonitorMetricsHandler.java:
```diff
@@ -0,0 +1,216 @@
+			mbs.addNotificationListener(MBeanServerDelegate.DELEGATE_NAME, listener, null, null);
+		} catch (InstanceNotFoundException e) {
+            if (tc.isDebugEnabled()) {
+                Tr.debug(tc, "getCount exception message: ", e.getMessage());
```
Unconditional debug statement

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/MonitorMetricsHandler.java:
```diff
@@ -0,0 +1,216 @@
+    		if (mm.getObjectName().equals(objectName)) {
+    			removeSet.add(mm);
+    			mm.unregisterMetrics(sharedMetricRegistry);
+    			Tr.debug(tc, "Monitoring MXBean " + objectName + " was unregistered from mpMetrics.");
```
Unconditional debug statement

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/MonitorMetricsHandler.java:
```diff
@@ -0,0 +1,216 @@
+                                break;
+                            }
+                        }
+                        registerMbeanObjects(mBeanObjectInstanceSetTemp);
```
When timed out, you will get an empty `mBeanObjectInstanceSetTemp`.  Do you need to handle that?

fmhwong commented:
Please fix indentation.  It is harder to read when the code doesn't have proper indentation.  Probably due to mix of tabs and spaces.

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/helper/Util.java:
```diff
@@ -0,0 +1,58 @@
+/*******************************************************************************
+ * Copyright (c) 2017, 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0.monitor/src/io/openliberty/microprofile/metrics/internal/monitor/package-info.java:
```diff
@@ -0,0 +1,18 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/helper/package-info.java:
```diff
@@ -0,0 +1,18 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/ApplicationListener30.java:
```diff
@@ -0,0 +1,83 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
2022

posmith commented:
L2 message review not required

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/BaseMetricsRESTProxyServlet.java:
```diff
@@ -0,0 +1,92 @@
+/*******************************************************************************
+ * Copyright (c) 2017 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/MetricRESTHandler.java:
```diff
@@ -0,0 +1,156 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/MetricsConfig.java:
```diff
@@ -0,0 +1,186 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/SharedMetricRegistries.java:
```diff
@@ -0,0 +1,106 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/SmallryeMetricsCDIMetadata.java:
```diff
@@ -0,0 +1,275 @@
+                classPath.add(micrometerJarFile);
+            } catch (FileNotFoundException e) {
+                // TODO: specific ERROR/WARNING?
+                Tr.debug(tc, e.getMessage());
```
Unconditional debug statement

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/SharedMetricRegistries.java:
```diff
@@ -0,0 +1,106 @@
+                metricRegistry.addNameToApplicationMap(metricID);
+            } catch (ClassCastException e) {
+                // This should never actually happen.
+                Tr.debug(tc, "Incompatible Metric Registries. Coud not cast " + registry + " to "
```
Unconditional debug statement

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/microprofile/metrics50/internal/package-info.java:
```diff
@@ -0,0 +1,18 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/adapters/SRMetricRegistryAdapter.java:
```diff
@@ -0,0 +1,83 @@
+
+    public void addNameToApplicationMap(MetricID metricID) {
+        try {
+            addNameToApplicationMapMethod.invoke(srMetricRegistryObject, metricID);
```
`addNameToApplicationMapMethod` could be `null`.

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/adapters/SRMetricRegistryAdapter.java:
```diff
@@ -0,0 +1,83 @@
+
+    public void unRegisterApplicationMetrics(String name) {
+        try {
+            unRegisterApplicationMetricsMethod_StringParam.invoke(srMetricRegistryObject, name);
```
`unRegisterApplicationMetricsMethod_StringParam` could  be `null`

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/adapters/SRSharedMetricRegistriesAdapter.java:
```diff
@@ -0,0 +1,126 @@
+
+        Object metricRegistryObject;
+        try {
+            metricRegistryObject = getOrCreateMethodAppNameResolver.invoke(null, scope,
```
`getOrCreateMethodAppNameResolver` could be `null`

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricProducerAdapter.java:
```diff
@@ -0,0 +1,146 @@
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "The SmallRye MetricProducer class was not resolved.");
+
+                // Use an Error message like the following?!?!
```
Is this still needed?

Channyboy replied:
The comment was left for discussion. Whether we use a debug statement if the class isn't loaded or throw a error message to be translated.

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.5.0/src/io/openliberty/smallrye/metrics/cdi/adapters/MetricRegistryProducerAdapter.java:
```diff
@@ -0,0 +1,85 @@
+        if (srMetricsProducerClass == null) {
+            if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled()) {
+                Tr.debug(tc, "The SmallRye MetricRegistryProduer class was not resolved.");
+                // Use an Error message like the following?!?!
```
Is this still needed?

Channyboy replied:
The comment was left for discussion. Whether we use a debug statement if the class isn't loaded or throw a error message to be translated.

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/.classpath:
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Should the project name be `io.openliberty.microprofile.metrics.5.0.internal.private`?

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/bnd.bnd:
```diff
@@ -0,0 +1,57 @@
+#*******************************************************************************
+# Copyright (c) 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/resources/WEB-INF/web.xml:
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright (c) 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/resources/io/openliberty/microprofile/metrics/internal/privateapi/resources/ibm-web-ext.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright (c) 2012 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/src/io/openliberty/microprofile/metrics/internal/privateapi/PrivateMetricsRESTHandler.java:
```diff
@@ -0,0 +1,61 @@
+/*******************************************************************************
+ * Copyright (c) 2018, 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/src/io/openliberty/microprofile/metrics/internal/privateapi/PrivateMetricsRESTProxyServlet.java:
```diff
@@ -0,0 +1,17 @@
+/*******************************************************************************
+ * Copyright (c) 2018 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.private.5.0/src/io/openliberty/microprofile/metrics/internal/privateapi/package-info.java:
```diff
@@ -0,0 +1,18 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.public.5.0/bnd.bnd:
```diff
@@ -0,0 +1,57 @@
+#*******************************************************************************
+# Copyright (c) 2021 IBM Corporation and others.
```
2022

fmhwong commented on dev/io.openliberty.microprofile.metrics.internal.public.5.0/resources/io/openliberty/microprofile/metrics/internal/publicapi/resources/ibm-web-ext.xml:
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+    Copyright (c) 2012 IBM Corporation and others.
```
2022

fmhwong reviewed (approved):
@Channyboy will have a follow up PR to address the review comments.

Channyboy commented:
#libby
